keras_contrib/applications/densenet.py

```diff
@@ -376,56 +385,46 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
         x = __conv_block(x, growth_rate, bottleneck, dropout_rate, weight_decay)
         x_list.append(x)
 
-        x1 = concatenate(x_list, axis=concat_axis)
-        #x = merge(x_list, mode='concat', concat_axis=concat_axis)
+        x = concatenate(x_list, axis=concat_axis)
 
         if grow_nb_filters:
             nb_filter += growth_rate
 
     if return_concat_list:
-        return x1, nb_filter, x_list
+        return x, nb_filter, x_list
     else:
-        return x1, nb_filter
+        return x, nb_filter
 
 
-def __transition_up_block(ip, nb_filters, type='upsampling', output_shape=None, weight_decay=1E-4):
+def __transition_up_block(ip, nb_filters, type='upsampling', weight_decay=1E-4):
     ''' SubpixelConvolutional Upscaling (factor = 2)
-
     Args:
         ip: keras tensor
         nb_filters: number of layers
-        type: can be 'upsampling', 'subpixel', 'deconv', or 'atrous'. Determines type of upsampling performed
-        output_shape: required if type = 'deconv'. Output shape of tensor
+        type: can be 'upsampling', 'subpixel', 'deconv'. Determines type of upsampling performed
         weight_decay: weight decay factor
-
     Returns: keras tensor, after applying upsampling operation.
     '''
 
     if type == 'upsampling':
         x = UpSampling2D()(ip)
     elif type == 'subpixel':
-        x = Conv2D(nb_filters, (3, 3), padding="same", kernel_regularizer=l2(weight_decay), activation='relu',
+        x = Conv2D(nb_filters, (3, 3), activation="relu", padding='same', W_regularizer=l2(weight_decay),
                    use_bias=False, kernel_initializer='he_uniform')(ip)
-        #x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
-        #                  bias=False, init='he_uniform')(ip)
         x = SubPixelUpscaling(scale_factor=2)(x)
-        x = Conv2D(nb_filters, (3, 3), activation="relu", padding='same', kernel_regularizer=l2(weight_decay),
```

Comment:
Is the removal of this second convolution in the subpixelupscaling case correct?

Also does the subpixel upscaling even work & is it necessary? The 100 layers paper describes its dense blocks after each upscaling step as essentially providing this functionality.